extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -248,43 +267,69 @@ export class AmpAutocomplete extends AMP.BaseElement {
    * @private
    */
   selectHandler_(event) {
-    if (!this.isItemElement_(event.target)) {
-      return Promise.resolve();
-    }
-    return this.mutateElement(() => {
-      this.selectItem_(event.target);
+    let item;
+    return this.measureMutateElement(() => {
+      item = this.getItemElement_(event.target);
+    }, () => {
+      this.selectItem_(item);
     });
   }
 
   /**
    * Render filtered results on the current input and update the container_.
+   * @return {!Promise}
    * @private
    */
   renderResults_() {
     const userInput = this.inputElement_.value;
     this.clearAllItems_();
     if (userInput.length < this.minChars_ || !this.inlineData_) {
-      return;
+      return Promise.resolve();
     }
+
     const filteredData = this.filterData_(this.inlineData_, userInput);
-    filteredData.forEach(item => {
-      this.container_.appendChild(this.createElementFromItem_(item));
+    let renderPromise = Promise.resolve();
+    if (this.templateElement_) {
+      renderPromise = this.templates_.renderTemplateArray(this.templateElement_,
+          filteredData).then(renderedChildren => {
+        renderedChildren.map(child => {
+          userAssert(child.hasAttribute('value'),
+              `${TAG} requires <template value=""> tag`);
```

Comment:
it's not really `<template value`, rather `first of template`

Reply:
It might be better to validate this earlier in the `buildCallback` ( e.g. if there is a template, make sure first child has value attribute`), no need to repeat validate in the rendering loop.